core/src/test/java/io/grpc/internal/DelayedStreamTest.java

```diff
@@ -154,11 +155,15 @@ public void setStream_cantCreateTwice() {
 
   @Test
   public void streamCancelled() {
-    stream.cancel(Status.CANCELLED);
+    Status status = Status.CANCELLED.withDescription("unique status message");
+    stream.cancel(status);
+    verify(listener).closed(same(status), isA(Metadata.class));
+    verifyNoMoreInteractions(listener);
 
-    // Should be a no op, and not fail due to transport not returning a newStream
+    // Should be _mostly_ a no op, and not fail due to transport not returning a newStream
     stream.setStream(realStream);
 
-    verify(listener).closed(eq(Status.CANCELLED), isA(Metadata.class));
+    verifyNoMoreInteractions(listener);
+    verify(realStream).cancel(same(status));
```

Comment:
When possible, I prefer same because it is more clear and doesn't depend on the equals implementation in Status, which may change. If the implementation _must_ use the same reference, then same seems to be a better match.